### PR TITLE
Package libnixmain and libnixcmd with Meson

### PR DESCRIPTION
**Pull request by Ericson2314**
# Motivation

#2503

# Context

Depends on #10973

# Priorities and Process

Add :+1: to [pull requests you find important](https://github.com/NixOS/nix/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc).

The Nix maintainer team uses a [GitHub project board](https://github.com/orgs/NixOS/projects/19) to [schedule and track reviews](https://github.com/NixOS/nix/tree/master/maintainers#project-board-protocol).

### CODE REVIEW

**Comment by Ericson2314** [comment]
@roberth OK! Recomputed the diff so this is one commit now!

File: src/libcmd/meson.build [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+readline_flavor = get_option('readline-flavor')
+if readline_flavor == 'editline'
+  editline = dependency('libeditline', 'editline', version : '>=1.14')
+  deps_private += editline
+elif readline_flavor == 'readline'
+  readline = dependency('readline')
+  deps_private += readline
+  configdata.set(
+    'USE_READLINME',
```
Comment (roberth) [inline src/libcmd/meson.build:48]:
```suggestion
    'USE_READLINE',
```

File: src/libcmd/meson.options [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  type: 'feature',
+  description: 'Enable Markdown rendering in the Nix binary (requires lowdown)',
+)
+
+option(
+  'readline-flavor',
+  type : 'combo',
+  choices : ['editline', 'readline'],
+  value : 'editline',
+  description : 'Which library to use for nice line editting with the Nix language REPL',
```
Comment (roberth) [inline src/libcmd/meson.options:14]:
```suggestion
  description : 'Which library to use for nice line editing with the Nix language REPL',
```

**Review by roberth — APPROVED** [review]

**Comment by Ericson2314** [comment]
Sorry I took some Lix code for this, forgot to credit probably @Qyriad in a co-authored-by.

**Comment by roberth** [comment]
I guess you could revert and revert-revert to represent this in git?

**Pull request merged**